src/middleware/validator.js

```diff
@@ -1,95 +1,31 @@
-
-  if (validations) {
-    Object.keys(validations).forEach(key => {
-      let rule = validations[`${key}`]
-      switch (rule.type) {
-        case 'string':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.string().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.string().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.string()
-          }
-          break
-
-        case 'number':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.number().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.number().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.number()
-          }
-          break
-
-        case 'boolean':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.boolean().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.boolean().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.boolean()
-          }
-          break
-
-        case 'array':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.array().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.array().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.array()
-          }
-          break
+const configurations = config.getConfig()
 
-        case 'object':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.object().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.object().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.object()
-          }
-          break
-
-        default:
-          throw new Error(`Validation rule type is not supported: ${rule.type}`)
-      }
-    })
-
-    if (schemaObject) {
-      return Joi.object(schemaObject)
-    }
-
-    throw new Error('Joi validation schema creation failed')
+const performValidation = (ctx, schema) => {
+  if (!schema) {
+    throw new Error(`No validation rules supplied`)
   }
 
-  throw new Error('No validation schema in file')
-}
+  if (!ctx || !ctx.request || !ctx.request.body) {
+    throw new Error(`Invalid request body`)
+  }
 
-const performValidation = (ctx, schema) => {
-  const {error, value} = schema.validate(ctx.request.body)
+  const ajv = new Ajv({
+    nullable: configurations.nullable
+  })
+  const valid = ajv.validate(schema, ctx.request.body)
 
-  if (error) {
-    throw new Error(`Validation execution failed: ${error.message}`)
+  if (!valid) {
+    throw new Error(`Validation failed: ${ajv.errorsText()}`)
   }
 }
 
-exports.validateInput = validationMap => async (ctx, next) => {
+exports.validateInput = schema => async (ctx, next) => {
   try {
     const schema = createValidationSchema(validationMap)
```

Comment:
@bradsawadye Where is this funciton declared?

Reply:
The function was removed, when refactoring was done.  The statement has been removed in this https://github.com/jembi/openhim-mediator-mapping/pull/21/commits/2e0b61d608e28895df6c36427b86f2a83cc9a510